dali/python/nvidia/dali/ops/_signatures.py

```diff
@@ -370,63 +371,130 @@ def _get_op(api_module, full_qualified_name: List[str]):
-            file_path = nvidia_dali_path / api / module_path / "__init__.pyi"
+        if module_path not in self._module_to_file:
+            file_path = self._nvidia_dali_path / self._api / module_path / "__init__.pyi"
             os.makedirs(os.path.dirname(file_path), exist_ok=True)
             open(file_path, "w").close()  # clear the file
             f = open(file_path, "a")
-            module_to_file[module_path] = f
+            self._module_to_file[module_path] = f
             f.write(_HEADER)
             full_module_nesting = [""] + module_nesting
-            submodules_dict = module_tree
+            # Find out all the direct submodules and add the imports
+            submodules_dict = self._module_tree
             for submodule in full_module_nesting:
                 submodules_dict = submodules_dict[submodule]
             direct_submodules = submodules_dict.keys()
             for direct_submodule in direct_submodules:
                 f.write(f"from . import {direct_submodule}\n")
 
             f.write("\n\n")
+        return self._module_to_file[module_path]
+
+    def close(self):
+        for _, f in self._module_to_file.items():
+            f.close()
+
+
+def gen_all_signatures(nvidia_dali_path, api):
+    """Generate the signatures for "fn" or "ops" api.
+
+    Parameters
+    ----------
+    nvidia_dali_path : Path
+        The path to the wheel pre-packaging to the nvidia/dali directory.
+    api : str
+        "fn" or "ops"
+    """
+    nvidia_dali_path = Path(nvidia_dali_path)
+
+    stub_manager = StubFileManager(nvidia_dali_path, api)
+
+    sig_groups = _group_signatures(api)
+
+    # Python-only and the manually defined ones are reexported from their respective modules
+    for (schema_name, op) in sig_groups["python_only"] + sig_groups["python_wrapper"]:
+        _, module_nesting, op_name = _names._process_op_name(schema_name, api=api)
+
+        stub_manager.get(module_nesting).write(f"\n\nfrom {op._impl_module} import"
+                                               f" ({op.__name__} as {op.__name__})\n\n")
+
+    # we do not go over sig_groups["hidden_or_internal"] at all as they are supposed to not be
+    # directly visible
+
+    # Runtime generated classes use fully specified stubs.
+    for (schema_name, op) in sig_groups["generated"]:
+        _, module_nesting, op_name = _names._process_op_name(schema_name, api=api)
+        schema = _b.TryGetSchema(schema_name)
 
         if api == "fn":
-            module_to_file[module_path].write(_gen_fn_signature(schema, schema_name, fn_name))
+            stub_manager.get(module_nesting).write(_gen_fn_signature(schema, schema_name, op_name))
         else:
-            module_to_file[module_path].write(_gen_ops_signature(schema, schema_name, op_name))
+            stub_manager.get(module_nesting).write(_gen_ops_signature(schema, schema_name, op_name))
 
-    for _, f in module_to_file.items():
-        f.close()
+    # TODO(klecki): use scope manager
```

Comment:
I think you can just wrap the `StubFileManager(nvidia_dali_path, api)` into `from contextlib import closing` like `with closing(StubFileManager(nvidia_dali_path, api)) as stub_manager`.

Reply:
done